test/e2e/pyrolisis_test.go

```diff
@@ -164,6 +168,48 @@ func (p *PyrolisisTests) CleanupBackupStorageContainers(t *testing.T) {
 	}
 }
 
+func (p *PyrolisisTests) CleanupSnapshots(t *testing.T) {
+	storageClient, err := openstack.NewBlockStorageV3(p.OpenStack.Provider, gophercloud.EndpointOpts{})
+	require.NoError(t, err, "There should be no error while creating storage client")
+
+	project, err := tokens.Get(p.OpenStack.Identity, p.OpenStack.Provider.Token()).ExtractProject()
+	require.NoError(t, err, "There should be no error while extracting the project")
+
+	listOpts := snapshots.ListOpts{
+		TenantID: project.ID,
+	}
+
+	allPages, err := snapshots.List(storageClient, listOpts).AllPages()
+	require.NoError(t, err, "There should be no error retieving all snapshot pages")
+
+	allSnapshots, err := snapshots.ExtractSnapshots(allPages)
+	require.NoError(t, err, "There should be no error extracting all snapshots")
+
+	var volumesInDeletionIds = make([]string, 1)
+	for _, snapshot := range allSnapshots {
+		err := snapshots.Delete(storageClient, snapshot.ID).ExtractErr()
+		require.NoError(t, err, "There should be no error deleting the snapshot")
+		volumesInDeletionIds = append(volumesInDeletionIds, snapshot.VolumeID)
+	}
+
+	// The storage client automatically deletes volumes alongside snapshots.
```

Comment:
What? If you delete a snapshot the volume is deleted as well? That does not make sense to me. What do you men by that.